examples/textual_inversion/textual_inversion.py

```diff
@@ -544,8 +569,10 @@ def main():
                 else:
                     grads = text_encoder.get_input_embeddings().weight.grad
                 # Get the index for tokens that we want to zero the grads for
-                index_grads_to_zero = torch.arange(len(tokenizer)) != placeholder_token_id
-                grads.data[index_grads_to_zero, :] = grads.data[index_grads_to_zero, :].fill_(0)
+                grad_mask = torch.arange(len(tokenizer)) != placeholder_token_ids[0]
+                for i in range(1, len(placeholder_token_ids)):
+                    grad_mask = grad_mask & (torch.arange(len(tokenizer)) != placeholder_token_ids[i])
+                grads.data[grad_mask, :] = grads.data[grad_mask, :].fill_(0)
```

Comment:
this masking logic has changed a bit as of https://github.com/huggingface/diffusers/pull/1665/files

I think you'll need something like:

```diff
  # Let's make sure we don't update any embedding weights besides the newly added token
- index_no_updates = torch.arange(len(tokenizer)) != placeholder_token_id
+ index_no_updates = torch.arange(len(tokenizer)) < placeholder_token_ids[0]
```

that's one of the changes I made when trying to manually rebase your changes onto my copy of the TI trainer from master:  
https://github.com/Birch-san/diffusers-play/commit/1d47955a3d827cb1de660287788c6719ede90ce5

Reply:
nice! Thanks for the comment will update and test in a bit!